atef/bin/config.py

```diff
@@ -0,0 +1,1528 @@
+                self.sev_on_failure_combo_items.index(
+                    self.bridge.severity_on_failure.get().name
+                )
+            )
+            self.if_disc_combo.setCurrentIndex(
+                self.if_disc_combo_items.index(
+                    self.bridge.if_disconnected.get().name
+                )
+            )
+            # Set up the generic item signals in order from top to bottom
+            self.invert_combo.currentIndexChanged.connect(
+                self.new_invert_combo
+            )
+            self.reduce_period_edit.textEdited.connect(
+                self.new_reduce_period_edit
+            )
+            self.reduce_method_combo.currentTextChanged.connect(
+                self.new_reduce_method_combo
+            )
+            self.string_combo.currentIndexChanged.connect(
+                self.new_string_combo
+            )
+            self.sev_on_failure_combo.currentTextChanged.connect(
+                self.new_sev_on_failure_combo
+            )
+            self.if_disc_combo.currentTextChanged.connect(
+                self.new_if_disc_combo
+            )
+            self.comparison_setup_done = True
+
+    def new_invert_combo(self, index: int):
+        self.bridge.invert.put(bool(index))
+
+    def new_reduce_period_edit(self, value: str):
+        try:
+            value = int(value)
+        except Exception:
+            value = 0
+        self.bridge.reduce_period.put(value)
+
+    def new_reduce_method_combo(self, value: str):
+        self.bridge.reduce_method.put(ReduceMethod[value])
+
+    def new_string_combo(self, index: int):
+        self.bridge.string.put(bool(index))
+
+    def new_sev_on_failure_combo(self, value: str):
+        self.bridge.severity_on_failure.put(Severity[value])
+
+    def new_if_disc_combo(self, value: str):
+        self.bridge.if_disconnected.put(Severity[value])
+
+
+def cast_dataclass(data: Any, new_type: type):
+    new_fields = dataclasses.fields(new_type)
+    new_kwargs = {
+        key: value for key, value in dataclasses.asdict(data).items()
+        if key in set(field.name for field in new_fields)
+    }
+    return new_type(**new_kwargs)
+
+
+class CompMixin:
+    data_type: ClassVar[type]
+
+    def __init_subclass__(cls, *args, **kwargs):
+        super().__init_subclass__(*args, **kwargs)
+        CompView.register_comparison(cls.data_type, cls)
+
+
+class ComparisonWidget(CompMixin, QLabel):
+    data_type = Comparison
+
+    def __init__(self, bridge: QDataclassBridge, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.setText('Please select a comparison type above.')
+
+
+# This class should be replaced by a real "Equals" widget
+class EqualsWidget(CompMixin, QLabel):
+    data_type = Equals
+
+    def __init__(self, bridge: QDataclassBridge, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.setText('This is just a placeholder for "Equals" for testing.')
+
+
+def main():
```

Comment:
`main` is where your argparse arguments would go, when `filename` (or `filenames`, perhaps) is added to the command-line options